src/credentials/MessageSecurity.cpp

```diff
@@ -0,0 +1,237 @@
+/*
```

Comment:
It's a little weird to have group decryption live here (with a pretty generic name) while unicast decryption lives under `src/transport`.

And `src/credentials` doesn't really seem like the right place for this code at first glance.... This directory is about certificat management, not message crypto operations.

Reply:
I agree, but I don't think `src/transport` is not the right place either. Since the GroupDataProvider has the trial `Decrypt` method, the encryption/decryption function should be made available for both `credentials` and `transport`. Also, the encryption/decryption functions should take the encryption key as input, so they may be used in both contexts.

Reply:
src/transport seems to be around low-level packet handling, so the encryption/decryption of packets seems to fit in there ok....  But I have to admit the organization of some of this stuff is not very clear to me, so maybe I'm missing something?

Yes, taking the key as input makes sense.